backend/internal/system/immutable_resource/composite_helpers.go

```diff
@@ -0,0 +1,418 @@
+func CompositeGetByFieldHelper[T any](
+	dbGetter func() (T, error),
+	fileGetter func() (T, error),
+	notFoundError error,
+) (T, error) {
+	var zero T
+
+	entity, err := dbGetter()
+	if err == nil {
+		return entity, nil
+	}
+
+	if errors.Is(err, notFoundError) {
+		entity, fileErr := fileGetter()
+		if fileErr == nil {
+			return entity, nil
+		}
+		return zero, notFoundError
+	}
+
+	return zero, err
+}
+
+// CompositeCreateHelper implements Create operation with immutable conflict checking.
+//
+// Type Parameter:
+//   - T: The domain entity type
+//
+// Parameters:
+//   - entity: The entity to create
+//   - getID: Function to extract ID from entity
+//   - fileGetter: Function to check if ID exists in file store
+//   - dbCreate: Function to create entity in database store
+//   - notFoundError: Domain-specific "not found" error to distinguish from other errors
+//
+// Returns:
+//   - Error if ID conflicts with immutable resource or creation fails
+//
+// Behavior:
+//  1. Extract ID from entity
+//  2. Check if ID exists in file store (immutable)
+//  3. If exists in file store (no error), return conflict error
+//  4. If not found (specific error), proceed with creation
+//  5. If other error (e.g., I/O, parsing), return that error
+//
+// This ensures you cannot create a runtime resource with the same ID
+// as an immutable resource, and properly handles file store errors.
+func CompositeCreateHelper[T any](
+	entity T,
+	getID func(T) string,
+	fileGetter func(string) (interface{}, error),
+	dbCreate func(T) error,
+	notFoundError error,
+) error {
+	id := getID(entity)
+
+	// Check if ID conflicts with immutable resource
+	_, err := fileGetter(id)
+	if err == nil {
+		// Resource exists in file store - conflict
+		return fmt.Errorf("resource with ID %s already exists as immutable resource", id)
+	}
+
+	// If error is not "not found", return it (could be I/O error, parsing error, etc.)
+	if !errors.Is(err, notFoundError) {
+		return err
+	}
+
+	// Resource not found in file store - safe to create in database
+	return dbCreate(entity)
+}
+
+// CompositeUpdateHelper blocks updates to immutable resources.
+//
+// Type Parameter:
+//   - T: The domain entity type
+//
+// Parameters:
+//   - entity: The entity to update
+//   - getID: Function to extract ID from entity
+//   - fileGetter: Function to check if ID exists in file store
+//   - dbUpdate: Function to update entity in database store
+//   - immutableError: Error to return when trying to update immutable resource
+//   - notFoundError: Domain-specific "not found" error to distinguish from other errors
+//
+// Returns:
+//   - Error if resource is immutable or update fails
+//
+// Behavior:
+//  1. Extract ID from entity
+//  2. Check if exists in file store
+//  3. If immutable (no error), return immutableError
+//  4. If not found (specific error), proceed with update
+//  5. If other error (e.g., I/O, parsing), return that error
+func CompositeUpdateHelper[T any](
```

Comment:
Same comment as `CompositeCreateHelper`, we don't need to have Composite Helper for update.